hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -578,7 +578,17 @@ private void initialize() {
     for (String walGroupId: logQueue.getQueues().keySet()) {
       tryStartNewShipper(walGroupId);
     }
-    this.startupOngoing.set(false);
+    setSourceStartupStatus(false);
+  }
+
+  private synchronized void setSourceStartupStatus(boolean initializing) {
+    if (initializing) {
```

Comment:
nit: to condense to fewer lines.

startupOnging.set(initializing);
initializing ? metrics.incrementSourceInitializing() : metrics.decr...